### PR TITLE
utils: added sshd config and service utilities

### PR DESCRIPTION
**Pull request by danlavu**

### CODE REVIEW

File: sssd_test_framework/utils/sshd.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            # Select authselect profile and feature
+            client.authselect.select("sssd", ["with-gssapi", "with-mkhomedir"])
+
+            # Start sssd
+            client.sssd.start()
+
+            # Configure ssh client and daemon
+            client.sshd.config_set([{"GSSAPIAuthentication": "yes"}])
+
+            # Start sshd
```
Comment (ikerexxe) [inline sssd_test_framework/utils/sshd.py:36]: I think this should be `Restart sshd`

**Review by ikerexxe — COMMENTED** [review]
Just a minor comment, everything else looks good.

File: sssd_test_framework/utils/sshd.py (hunk 1 above)
Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:36]: Yes, yes, it should.

**Review by ikerexxe — DISMISSED** [review]
LGTM! Thanks for the patch

File: sssd_test_framework/utils/sshd.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        self.args: str = f'--transform "sshd.lns incl {self.file}"'
+        self.cmd: str = ""
+
+    def setup_when_used(self) -> None:
+        super().setup_when_used()
+        self.fs.backup(self.file)
+
+    def config_read(self) -> str:
+        """
+        Read SSH daemon configuration as Augeas tree.
```
Comment (pbrezina) [inline sssd_test_framework/utils/sshd.py:69]: Add blank line. Same on other places.

**Review by pbrezina — CHANGES_REQUESTED** [review]
Hi Dan, please, see minor change in the doc string. Otherwise LGTM. I don't plan to test it, I trust that the code works for you.

**Comment by danlavu** [comment]
It works for me. Moving to approved.

**Comment by pbrezina** [comment]
@danlavu Remove the merge commit please.

**Comment by danlavu** [comment]
> @danlavu Remove the merge commit please.

Done.

File: sssd_test_framework/utils/sshd.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        Reload the SSH daemon.
+
+        :param service: Service to start, defaults to 'sshd'
+        :type service: str, optional
+        :param raise_on_error: Raise exception on error, defaults to True
+        :type raise_on_error: bool, optional
+        :return: SSH process result.
+        :rtype: SSHProcessResult
+        """
+
```
Comment (pbrezina) [inline sssd_test_framework/utils/sshd.py:128]: This blank line should not be present.

File: sssd_test_framework/utils/sshd.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        self.svc: SystemdServices = svc
+
+        self.file: str = file
+        self.path: str = "/files" + self.file
+        self.args: str = f'--transform "sshd.lns incl {self.file}"'
+        self.cmd: str = ""
+
+    def setup_when_used(self) -> None:
+        super().setup_when_used()
+        self.fs.backup(self.file)
```
Comment (pbrezina) [inline sssd_test_framework/utils/sshd.py:65]: I was about to push it, but this also requires a teardown to reload original configuration. Something like:

```
def teardown_when_used(self) -> None:
    self.fs.restore(self.file)
    self.svc.reload("sshd")
```

**Review by pbrezina — CHANGES_REQUESTED** [review]
Also how is this going to be used? Wouldn't it be maybe better to create new instance of sshd listening on different port so you don't meddle with ssh connections that we use to talk to the host?

File: sssd_test_framework/utils/sshd.py (hunk 4 above)
Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:65]: I thought pytest-mh , when using self.fs.backup restored the file. This needs to be updated for the Pam util class I wrote to then.

Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:66]: Moving to draft, something isn't right.

Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:66]: Okay, self.fs.restore(self.file) isn't necessary, the file does get restored.

Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:66]: https://pastebin.com/TiNs0Q3S

I put a print after the config_set and another in the teardown, the changes have already been reverted.

Comment (danlavu) [reply sssd_test_framework/utils/sshd.py:66]: I don't for see many tests using this function at all, this is mainly to enable GSSAPIAuthentication on the host which can be done safely with little impact. I updated the comment with a warning when using this. There're only a few tests that need to edit the sshd config.

Comment (pbrezina) [reply sssd_test_framework/utils/sshd.py:66]: Yes, at least for the moment.



> Okay, self.fs.restore(self.file) isn't necessary, the file does get restored.

This is something we need to work on in pytest-mh. To build a dependency tree so it is clear in which order are the MultihostUtility objects destroyed. But it works for us at the moment.

**Review by pbrezina — CHANGES_REQUESTED** [review]

**Pull request merged**